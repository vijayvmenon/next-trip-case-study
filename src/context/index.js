import { useContext } from 'react';
import AppContext, { Context } from './context';

export default AppContext;
export { Context };
export const useMyContext = () => useContext(Context);
