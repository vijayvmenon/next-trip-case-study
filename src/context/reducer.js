import constants from '../constants';
import ENTranslations from '../locales/en.json';
import { localStorageExists } from '../utils';

const localStorageLanguage =
  localStorageExists && localStorage.getItem('LANGUAGE');
const initialState = {
  tableExpanded: false,
  hideHeader: false,
  snackbar: { value: '', text: '' },
  language: localStorageLanguage || 'en',
  translations: localStorageLanguage
    ? constants.LANGUAGE_MAPPING[localStorageLanguage]
    : ENTranslations,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case 'TABLE_EXPAND':
      return { ...state, tableExpanded: true };
    case 'TABLE_COLLAPSE':
      return { ...state, tableExpanded: false };
    case 'SET_SNACKBAR':
      return { ...state, snackbar: { value: action.value, text: action.text } };
    case 'HIDE_HEADER':
      return { ...state, hideHeader: true };
    case 'SHOW_HEADER':
      return { ...state, hideHeader: false };
    case 'SET_LANGUAGE':
      return {
        ...state,
        language: action.value,
        translations: constants.LANGUAGE_MAPPING[action.value],
      };
    default:
      return { ...state };
  }
}

export { initialState };
