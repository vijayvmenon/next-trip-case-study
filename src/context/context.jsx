import React, { useReducer, createContext } from 'react';
import PropTypes from 'prop-types';
import reducer, { initialState } from './reducer';

export const Context = createContext();

export default function AppContext({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
}

AppContext.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,
};
