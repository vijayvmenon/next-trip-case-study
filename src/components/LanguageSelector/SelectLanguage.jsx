/* eslint-disable */
import React, { useContext } from 'react';
import { createUseStyles } from 'react-jss';
import { useLocation } from 'wouter';
import { Context } from '../../context';
import constants, { colors } from '../../constants';
import Select from '../UI/Select';
import { localStorageExists } from '../../utils';

const { LANGUAGE_SELECT } = constants;

const useStyles = createUseStyles({
  root: {
    background: 'white',
    borderRadius: 10,
    marginLeft: 'auto',
    marginRight: '1rem',
  },
});
export default function SelectLanguage() {
  const classes = useStyles();
  const { dispatch, state: { language = 'en' } = {} } = useContext(Context);
  const [, setLocation] = useLocation();

  const onSearch = (selected, label, id) => {
    // set localstorage LANGUAGE variable to persist the selection for browser reload
    if (localStorageExists) {
      localStorage.setItem('LANGUAGE', selected[id]);
    }
    dispatch({ type: 'SET_LANGUAGE', value: selected[id] });
  };

  return (
    <div className={classes.root}>
      <Select
        label={LANGUAGE_SELECT.label}
        onSearch={onSearch}
        options={LANGUAGE_SELECT.options}
        labelSelected={LANGUAGE_SELECT.labelSelected}
        idField={LANGUAGE_SELECT.idField}
        valueField={LANGUAGE_SELECT.valueField}
        selectedValue={language}
        languageDropdown={true}
      />
    </div>
  );
}
