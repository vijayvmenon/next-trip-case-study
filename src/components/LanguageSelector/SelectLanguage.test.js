import { render } from '@testing-library/react';
import SelectLanguage from './SelectLanguage';
import renderer from 'react-test-renderer';
import { Context } from '../../context';

const dispatch = jest.fn(); // mock dispatch

describe('Component:SelectLanguage.jsx', () => {
  test('SelectLanguage.jsx renders correctly', () => {
    const selectLanguage = renderer
      .create(
        <Context.Provider value={{ dispatch }}>
          <SelectLanguage />
        </Context.Provider>,
      )
      .toJSON();
    expect(selectLanguage).toMatchSnapshot();
  });
});
