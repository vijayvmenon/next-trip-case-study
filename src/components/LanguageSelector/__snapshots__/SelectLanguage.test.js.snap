// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:SelectLanguage.jsx SelectLanguage.jsx renders correctly 1`] = `
<div
  className="root-0-2-1"
>
  <div
    className="rootLanguage-0-2-3"
  >
    <select
      className="languageSelect-0-2-10 languageSelected-0-2-5"
      data-testid="custom-select-id"
      onChange={[Function]}
      value="en"
    >
      <option
        data-testid="option-default-id"
        disabled={true}
        hidden={true}
        value="Select Language"
      />
      <option
        className="selectedOption-0-2-12"
        data-testid="option-en-id"
        value="en"
      >
        English
      </option>
      <option
        data-testid="option-es-id"
        value="es"
      >
        Spanish
      </option>
    </select>
  </div>
</div>
`;
